apps/dashboard/src/components/activity/hooks/use-activity-by-transaction.ts

```diff
@@ -0,0 +1,39 @@
+import { useFetchActivities } from '@/hooks/use-fetch-activities';
+import { useEffect, useState } from 'react';
+
+interface UseActivityByTransactionProps {
+  transactionId?: string;
+  onActivityFound: (activityId: string) => void;
+}
+
+export function useActivityByTransaction({
```

Comment:
This hook is needed to correct a misalignment in the dashboard's Activity Feed. The current notification item, `? activityItemFeed=...`, is already present in the address bar. 

So, it's unclear if we rerun based on transactionID or Notification ID. 

<img width="1512" alt="Screenshot 2025-01-21 at 12 42 34" src="https://github.com/user-attachments/assets/18f1e836-b40c-423e-be1e-7db3716bb54c" />

My suggestion is to use the `notificationId` for now. We shouldn't need the useState; we should get it from the URL. That would eliminate the need for this hook.

Having said that, I suggest changing the URL schema to 
`https://dashboard-v2.novu.co/env/:env_id/activity-feed/:notificationId` to map with the API powering this screen.

Reply:
Sorry, but I do not entirely follow your suggestion. The re run uses the current notification entity, and performs a trigger using the same payload and recipient properties as the previous trigger. 

useActivityByTransactionId is just a refactor I did that encapsulates the fetch activities given a specific transactionId, this happens in 2 cases:
- We perform a trigger in the test workflow sidebar (Workflow editor), the API returns a transactionId, and we need to obtain the notification Id by searching the notification feed based on the transaction id and select the first activity item.
- When we re-run an activity item, the API returns us a transactionId, which we again need to convert to a notificationId/activityId to display the full details.